hail/python/hail/backend/backend.py

```diff
@@ -78,8 +80,15 @@ def remove_liftover(self, name, dest_reference_genome):
     def parse_vcf_metadata(self, path):
         pass
 
+    @property
+    @abc.abstractmethod
+    def fs(self):
+        pass
+
 
 class SparkBackend(Backend):
+    fs = HadoopFS()
```

Comment:
Hmm, this seems a bit weird.  I would have expected it in the constructor.

Reply:
I couldn't find a way to get abc to express abstract instance attributes.
  * A brief look around SO: https://stackoverflow.com/questions/51055750/python-abstract-instance-variable

"TypeError: Can't instantiate abstract class SparkBackend with abstract methods fs"
```
#... rest of FS ...
    @property
    @abc.abstractmethod
    def fs(self):
        pass

class SparkBackend(Backend):
    def __init__(self):
        self.fs = HadoopFS()
```

<img width="733" alt="Screenshot 2019-04-15 23 47 34" src="https://user-images.githubusercontent.com/5543229/56180803-dbfee300-5fd8-11e9-9fea-159e5a64c4b1.png">

Reply:
if you define 
```
class SparkBackend(Backend):
    def __init__(self):
        self._fs = HadoopFS()

    @property
    def fs(self):
        return self._fs
```

does it work?

Reply:
What @tpoterba said.  You can't implement a property with an attribute.

Reply:
Yep, that works, thanks!